src/cheevos/Cheevos.cpp

```diff
@@ -91,6 +98,102 @@ void CCheevos::EvaluateRichPresence(char* evaluation, size_t size)
   rc_evaluate_richpresence(m_richPresence, evaluation, size, PeekInternal, this, NULL);
 }
 
+void CCheevos::ActivateAchievement(unsigned cheevo_id, const char* memaddr) //1
+{
+  int res = rc_runtime_activate_achievement(&m_runtime, cheevo_id, memaddr, NULL, 0);
+  if (res == 0)
+  {
+    kodi::Log(ADDON_LOG_ERROR, "%u", cheevo_id);
+  }
+
+
+  // it will return integer value 0 in case achivement is activated successfully.
+}
+
+bool CCheevos::AwardAchievement(char* url,
+                                size_t size,
+                                const char* username,
+                                const char* token,
+                                unsigned cheevo_id,
+                                int hardcore,
+                                const char* game_hash)
+{
+
+  return rc_url_award_cheevo(url, size, username, token, cheevo_id, 0, game_hash) >= 0;
+}
+
+bool CCheevos::GetCheevo_ID(void (*Callback)(char* achievement_url, unsigned cheevo_id))
+{
+  //kodi::Log(ADDON_LOG_ERROR, "Fine till nowwwww in Callback");
+  Callback(url, m_cheevo_id);
+  return true;
+}
+
+void CCheevos::DeactivateTriggeredAchievement(unsigned cheevo_id)
```

Comment:
you could make this method to return bool and do something like the other methods